_episodes/02-keras.md

```diff
@@ -59,179 +59,204 @@ In this episode we will focus on a minimal example for each of these steps, late
+> > A few possible comments:
+> > - Columns `object_diameter` and `object_height` can be good features.
+> > - Columns `ceramics_image_type` and `ceramics_mark` for example are not good features due to very high number of NaNs.
+> > - Columns `start_date` and `end_date` are do not make good features as they are not related to the classification we want to achieve.
 > {:.solution}
 {:.challenge}
 
-### Input and Output Selection
-Now that we have familiarized ourselves with the dataset we can select the data attributes to use
-as input for the neural network and the target that we want to predict.
-
-In the rest of this episode we will use the `bill_length_mm`, `bill_depth_mm`, `flipper_length_mm`, `body_mass_g` attributes.
-The target for the classification task will be the `species`.
+In the rest of this episode we will use the `object_diameter`, `object_height`, `material_simplified` attributes.
+The target for the classification task will be the `l2_class`.
 
 > ## Data Exploration
 > Exploring the data is an important step to familiarize yourself with the problem and to help you
 > determine the relevant inputs and outputs.
 {:.keypoints}
 ## 3. Prepare data
-The input data and target data are not yet in a format that is suitable to use for training a neural network.
 
-### Change types if needed
-First, the species column is our categorical target, however pandas still sees it as the
-generic type `Object`. We can convert this to the pandas categorical type:
+### Remove unnecessary columns of data
+The dataset currently contains a lot of redundant or unnecessary data columns. We will remove all columns except our input and output columns. 
 ~~~
-penguins['species'] = penguins['species'].astype('category')
+ds_preprocessed = ds.drop(columns=['find_number', 'material', 'start_date',
+       'end_date', 'ceramics_image_type', 'ceramics_mark', 'on_website', 'url'])
 ~~~
 {:.language-python}
-This will make later interaction with this column a little easier.
+
+The input data and target data are not yet in a format that is suitable to use for training a neural network.
 
 ### Clean missing values
-During the exploration phase you may have noticed that some rows in the dataset have missing (NaN)
+During the exploration phase we saw that some rows in the dataset have missing (NaN)
 values, leaving such values in the input data will ruin the training, so we need to deal with them.
 There are many ways to deal with missing values, but for now we will just remove the offending rows by adding a call to `dropna()`:
 ~~~
-# Drop two columns and the rows that have NaN values in them
-penguins_filtered = penguins.drop(columns=['island', 'sex']).dropna()
+# Drop the rows that have NaN values in them
+ds_preprocessed = ds_preprocessed.dropna()
+~~~
+{:.language-python}
 
-# Extract columns corresponding to features
-penguins_features = penguins_filtered.drop(columns=['species'])
+> ## Pairplot: Visual Aid
```

Comment:
I opted to place it in a callout box because it did not necessarily go with the flow of the chapter but I did want to show them what a pairplot is and how it works.